hbase-hbtop/src/main/java/org/apache/hadoop/hbase/hbtop/Record.java

```diff
@@ -0,0 +1,180 @@
+    private FieldValue value;
+
+    private Entry(Field key, FieldValue value) {
+      this.key = Objects.requireNonNull(key);
+      this.value = Objects.requireNonNull(value);
+    }
+
+    @Override
+    public Field getKey() {
+      return key;
+    }
+
+    @Override
+    public FieldValue getValue() {
+      return value;
+    }
+
+    @Override
+    public FieldValue setValue(FieldValue value) {
+      FieldValue oldValue = this.value;
+      this.value = value;
+      return oldValue;
+    }
+  }
+
+  public static Entry entry(Field field, Object value) {
+    return new Entry(field, field.newValue(value));
+  }
+
+  public static Entry entry(Field field, FieldValue value) {
+    return new Entry(field, value);
+  }
+
+  public static Record ofEntries(List<Entry> entries) {
+    return ofEntries(entries.stream());
+  }
+
+  public static Record ofEntries(Entry... entries) {
+    return ofEntries(Stream.of(entries));
+  }
+
+  public static Record ofEntries(Stream<Entry> entries) {
+    return entries
+      .collect(Record::new, (r, e) -> r.put(e.getKey(), e.getValue()), (r1, r2) -> {});
+  }
+
+  public Record() {
+    this(new EnumMap<>(Field.class));
+  }
+
+  private Record(Map<Field, FieldValue> values) {
+    this.values = values;
+  }
+
+  @Override
+  public int size() {
+    return values.size();
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return values.isEmpty();
+  }
+
+  @Override
+  public boolean containsKey(Object key) {
+    return values.containsKey(key);
+  }
+
+  @Override
+  public boolean containsValue(Object value) {
+    return values.containsValue(value);
+  }
+
+  @Override
+  public FieldValue get(Object key) {
+    return values.get(key);
+  }
+
+  @Override
+  public FieldValue put(Field key, FieldValue value) {
+    return values.put(key, value);
+  }
+
+  public FieldValue put(Field key, Object value) {
+    return values.put(key, key.newValue(value));
+  }
+
+  @Override
+  public FieldValue remove(Object key) {
+    return values.remove(key);
+  }
+
+  @Override
+  public void putAll(@NotNull Map<? extends Field, ? extends FieldValue> m) {
+    values.putAll(m);
+  }
+
+  @Override
+  public void clear() {
+    values.clear();
+  }
+
+  @Override
+  @NotNull
+  public Set<Field> keySet() {
+    return values.keySet();
+  }
+
+  @Override
+  @NotNull
+  public Collection<FieldValue> values() {
+    return values.values();
+  }
+
+  @Override
+  @NotNull
+  public Set<Map.Entry<Field, FieldValue>> entrySet() {
+    return values.entrySet();
+  }
+
+  public Record combine(Record o) {
+    return ofEntries(values.keySet().stream()
+      .map(k -> {
+        switch (k.getFieldValueType()) {
+          case STRING:
+            return entry(k, values.get(k));
+          default:
+            return entry(k, values.get(k).plus(o.values.get(k)));
+        }
+      }));
+  }
+
+  public Record toImmutable() {
+    return new Record(Collections.unmodifiableMap(values));
+  }
+}
```

Comment:
Nice. Clean. Generic.  Is there a lib that has this already I wonder?

Reply:
Back here again. This is good. I like how it implements Map.

Reply:
I think we can use ImmutableMap in Guava here. Will modify to use it.